### PR TITLE
BREAKING: Remove namespaced permissions

### LINKED ISSUE #1323: Remove namespaced permissions from the `PermissionController`
Namespaced permissions were introduced as part of the original `PermissionController` (`rpc-cap`) because we thought it would be a good way to parameterize certain permissions. For example, you could do something like `wallet_deriveKey_*` where `*` is a BIP-44 derivation path. We did all of this and more at various points during Snaps development.

Eventually, we realized that anything expressible as a namespaced permission is also expressible as a caveat. Furthermore, caveats are no more difficult to make intelligible to the user in the UI than a namespaced permission.

Ultimately, by having both namespaced permissions and caveats in our permission system, we have two different ways of parameterizing / attenuating permissions, to no obvious benefit. While everything expressible as a namespaced permission is expressible as a caveat, the reverse is not true (see e.g. `eth_accounts`). For this reason, we should remove namespaced permissions from the `PermissionController`.

### PR DESCRIPTION
**Pull request by rekmarks**
## Description

Removes namespaced permissions from the `PermissionController` for the reasons described in #1323. As a consequence, also removes the notion of "target keys", which only existed to support namespaced methods. Controller state is not impacted by these changes.

## Changes

- **BREAKING**: Remove namespaced permissions
  - Namespaced permissions are no longer supported. Consumers should replace namespaced permissions with equivalent caveat-based implementations.
- **BREAKING**: Remove `targetKey` concept
  - The target key/name distinction only existed to support namespaced permissions, which are removed as of this release. Henceforth, permissions only have "names".
  - The `targetKey` property of permission specifications has been renamed to `targetName`.

## References

Fixes #1323 

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/permission-controller/src/PermissionController.ts [hunk 1]
```diff
@@ -1477,88 +1472,49 @@ export class PermissionController<
       permission.caveats.splice(caveatIndex, 1);
     }
 
-    this.validateModifiedPermission(permission, origin, target);
+    this.validateModifiedPermission(permission, origin);
```
Comment (FrederikBolding) [inline packages/permission-controller/src/PermissionController.ts:1475]: Seems like we could remove `target` from the arguments of `deleteCaveat`? Not sure why it needs to be specified when we have the permission object.

Comment (FrederikBolding) [reply packages/permission-controller/src/PermissionController.ts:1475]: Unless there is a good reason I am missing.

**Review by FrederikBolding — COMMENTED** [review]
Overall this looks great - just a few small comments.

File: packages/permission-controller/src/PermissionController.ts (hunk 1 above)
Comment (rekmarks) [reply packages/permission-controller/src/PermissionController.ts:1470]: Done in: [09c3171](https://github.com/MetaMask/core/pull/1337/commits/09c3171cbe0ca95266a6a5b8953e96e2c60bca3a)

**Review by FrederikBolding — APPROVED** [review]
LGTM

**Pull request merged**